source/app.d

```diff
@@ -210,6 +298,93 @@ void updateGithubComment(string action, IssueRef[] refs, Issue[] descs, string c
+alias LabelInfo = Tuple!(bool, "hasAutoMerge", bool, "hasAutoMergeSquash");
+
+auto analyseLabels(Json[] labels)
+{
+    auto labelNames = labels.map!`a["name"].get!string`.array;
+    bool hasAutoMerge = labelNames.any!`a == "auto-merge"`;
+    bool hasAutoMergeSquash = labelNames.any!`a == "auto-merge-squash"`;
+    return LabelInfo(hasAutoMerge, hasAutoMergeSquash);
+}
+
+auto handleGithubLabel(string repoSlug, uint pullRequestNumber, string commitsURL, bool isOpen)
+{
+    auto url = "%s/repos/%s/issues/%d/labels".format(githubAPIURL, repoSlug, pullRequestNumber);
+    auto res = ghGetRequest(url);
+
+    auto labels = res.readJson[];
+    auto labelInfo = analyseLabels(labels);
+    Json[] commits = null;
+
+    if (labelInfo.hasAutoMerge || labelInfo.hasAutoMergeSquash)
+    {
+        commits = getCommits(commitsURL);
+
+        if (!isOpen)
+        {
+            logWarn("Can't auto-merge PR %s/%d - it is already closed", repoSlug, pullRequestNumber);
+            goto end;
+        }
+
+        if (commits.length == 0)
+        {
+            logWarn("Can't auto-merge PR %s/%d has no commits attached", repoSlug, pullRequestNumber);
+            goto end;
+        }
+
+        string lastCommitSha = commits[$ - 1]["sha"].get!string;
+        string[string] reqInput = ["sha": lastCommitSha];
+
+        // lazy implementation -> try merge
+        if (labelInfo.hasAutoMerge)
+            reqInput["merge_method"] = "merge";
+
+        if (labelInfo.hasAutoMergeSquash)
+            reqInput["merge_method"] = "squash";
+
+        auto prUrl = "%s/repos/%s/pulls/%d/merge".format(githubAPIURL, repoSlug, pullRequestNumber);
+        ghSendRequest((scope req){
+            req.method = HTTPMethod.PUT;
+            // custom media type is required during preview period:
+            // https://developer.github.com/changes/2016-09-26-pull-request-merge-api-update/
+            req.headers["Accept"] = "application/vnd.github.polaris-preview+json";
+            req.writeJsonBody(reqInput);
+        }, prUrl);
+    }
+    end:
+    return LabelsAndCommits(labels, commits);
+}
+
+void checkAndRemoveMergeLabels(Json[] labels, string repoSlug, uint pullRequestNumber)
+{
+    auto labelInfo = analyseLabels(labels);
+    if (labelInfo.hasAutoMerge || labelInfo.hasAutoMergeSquash)
+    {
+        auto labelUrl = "%s/repos/%s/issues/%d/labels/".format(githubAPIURL, repoSlug, pullRequestNumber);
+        if (labelInfo.hasAutoMerge)
+            labelUrl ~= "auto-merge";
+        if (labelInfo.hasAutoMergeSquash)
+            labelUrl ~= "auto-merge-squash";
+        ghSendRequest(HTTPMethod.DELETE, labelUrl);
+    }
+}
+
+void searchForAutoMergePrs(string repoSlug)
+{
+    auto prs = ghGetRequest("%s/repos/%s/pulls?state=open".format(githubAPIURL, repoSlug)).readJson[];
+    foreach (pr; prs)
+    {
+        handleGithubLabel(repoSlug, pr["number"].get!uint, pr["commits_url"].get!string, true);
```

Comment:
> Not restricting label?
Sounds a bit better to first get all label:auto-merge, then all label:auto-merge-squash, subtract the latter from the former (in case of both), and then loop over the 2 arrays.

At GitHub there are two APIs for this:  `issue` and `pulls`.
With the `issue` API we can directly filter on the labels -> #17